### ISSUE TITLE
[Question] Are parameters supported for ALTER TABLE statements?

### ISSUE THREAD

**Issue by Endominus** [rusqlite/rusqlite#1175]
I've been trying to execute the below command for a couple hours now, using named parameters and positional parameters, encapsulating the parameters in parentheses as some of the documentation seems to do and not. 

```sql
ALTER TABLE entries ADD COLUMN :name :type;
```
```rust
self.conn.execute("ALTER TABLE entries ADD COLUMN :name :type;", &[(":name", field.name.as_str()), (":type", type_full)]);
```

I get a syntax error at the :name (or opening parenthesis if using that). I've tried executing using the execute function and preparing a statement to execute separately. Given that similar code works with the `INSERT` command, and when hardcoding the name and type above, I can only guess that there's some issue in executing `ALTER TABLE` with parameters but there doesn't seem to be any note of this in the documentation. It's a statement, per the sqlite docs, and there doesn't seem to be a different specialized statement for table-altering commands in the rusqlite docs as far as I found.

Am I just blind and missing something obvious? Thanks in advance!

**Comment by thomcc** [comment]
Nope, this is not supported. One way of thinking of this is that the value parameters evaluate are always expressions (in this case one evaluating to a string), and not identifiers. Whereas the column name must be an identifier -- perhaps you can think of it analgously to `let "foo" = 3`.

(That said, this explanation is somewhat handwavey and is not *really* accurate -- the true answer is probably just that it's a syntax error, and I'm making a guess to why that might be)
